### PR TITLE
dry edit role modal

### PR DESCRIPTION
**Pull request by stephherbers**
##### SUMMARY
Modified edit_role_modal.html to add a template for Access Areas and Report sections.

https://trello.com/c/bzrYuuZ7/103-dry-up-editrolemodalhtml

### CODE REVIEW

**Review by calellowitz — CHANGES_REQUESTED** [review]
Woohoo first PR and net 200 negative is a strong start! Few small things but looks good

File: corehq/apps/users/static/users/js/roles.js [hunk 1]
```diff
@@ -72,6 +72,159 @@ hqDefine('users/js/roles',[
...
+                self.accessAreas = [
+                {
+                    editPermission: self.permissions.edit_web_users,
+                    viewPermission: self.permissions.view_web_users,
+                    text:  "<strong>Web Users</strong> &mdash; invite new web users, manage account settings, remove membership",
+                    showEditCheckbox: true,
+                    editCheckboxLabel: "edit-web-users-checkbox",
+                    showViewCheckbox: true,
+                    viewCheckboxLabel: "view-web-users-checkbox",
+                    srEditAndViewText: gettext("Edit & View Mobile Workers"),
```
Comment (orangejenny) [inline corehq/apps/users/static/users/js/roles.js:84]: `sr` isn't a common prefix in our js. What would you think of renaming this to either `screenReaderEditAndViewText` or just `editAndViewText`?

**Review by orangejenny — COMMENTED** [review]
What beautiful code deletion. Looks good, pending Cal's feedback.

File: corehq/apps/users/templates/users/partials/edit_role_modal.html [hunk 2]
```diff
@@ -41,512 +43,89 @@ <h4 class="modal-title" data-bind="text: modalTitle"></h4>
...
-                         data-bind="checked: permissions.view_roles, disable: !$root.allowEdit" />
-                  <label for="view-roles-checkbox">
-                    <span class="sr-only">
-                      {% trans "View Roles and Permssions" %}
-                    </span>
-                  </label>
-                </div>
+                  <div data-bind="html: text"></div>
+                  {# PLACEHOLDER checkbox when edit_commcare_users and edit_groups is true #}
+                  <div class="form-check" data-bind="visible: $parent.permissions.edit_commcare_users() && editPermission() && showAllowCheckbox">
```
[the remark this answers is not in the document]
Comment (stephherbers) [reply corehq/apps/users/templates/users/partials/edit_role_modal.html:89]: Confirmed that is is true for both of the inner checkboxes

File: corehq/apps/users/static/users/js/roles.js [hunk 3]
```diff
@@ -72,6 +72,200 @@ hqDefine('users/js/roles',[
...
+                        screenReaderEditAndViewText: gettext("Edit & View Apps"),
+                        screenReaderViewOnlyText: null,
+                        showAllowCheckbox: false,
+                        allowCheckboxText: null,
+                        allowCheckboxId: null,
+                        allowCheckboxPermission: null,
+                    },
+                    {
+                        showOption: self.permissions.access_all_locations,
+                        editPermission: function () {return false},
```
Comment (calellowitz) [inline corehq/apps/users/static/users/js/roles.js:206]: Does this work in the places that refer to `editPermission` rather than `editPermission()`

**Review by calellowitz — COMMENTED** [review]
Few small things that aren't blocking and one question. I think stickler is right as well

File: corehq/apps/users/static/users/js/roles_and_permissions.js [hunk 4]
```diff
@@ -66,7 +66,9 @@ hqDefine("users/js/roles_and_permissions",[
             allowEdit: initialPageData.get("can_edit_roles"),
             canRestrictAccessByLocation: initialPageData.get("can_restrict_access_by_location"),
             landingPageChoices: initialPageData.get("landing_page_choices"),
-            webAppsPrivilege: initialPageData.get("web_apps_privilege")
+            webAppsPrivilege: initialPageData.get("web_apps_privilege"),
+            DataFileDownloadEnabled: initialPageData.get("data_file_download_enabled"),
```
Comment (calellowitz) [inline corehq/apps/users/static/users/js/roles_and_permissions.js:70]: nit: these should be lowerCamelCase

**Review by calellowitz — APPROVED** [review]
One nit that isn't blocking but would be good to fix in follow up

**Comment by esoergel** [comment]
Hi @stephherbers, it looks there's an issue with the scoping in `roles.js`, which makes the dropzone permission stay hidden.  It should appear when that feature flag is enabled, like this:

image

However, it only displays if `self.DataFileDownloadEnabled == true`.  That's set a bit lower in the file:

https://github.com/dimagi/commcare-hq/blob/0ae9226a6ea37e7aea13461b2f4cb90738bb5902/corehq/apps/users/static/users/js/roles.js#L307

That's a different `self`, though, so that variable is out of scope

**Comment by calellowitz** [comment]
This had some incorrect logic that caused the two permissions controlled by feature flags to never appear. Fix is https://github.com/dimagi/commcare-hq/pull/28447

**Pull request merged**